[Varatharaju Vignesh] iP

Vignesh-30 opened the pull request.

Vignesh-30 committed:
Level 0. Greet added

Vignesh-30 committed:
Seperated Level 0 into Greet and Bye methods

Vignesh-30 committed:
Added indent and improved lines

Vignesh-30 committed:
Level 1 added

Vignesh-30 committed:
Level 2 added

Vignesh-30 committed:
Added Task class.

Vignesh-30 committed:
Added level 3.

Vignesh-30 committed:
Changed variables from private to protected in Task class.

Vignesh-30 committed:
Revert "Changed variables from private to protected in Task class."

This reverts commit 0b90521c93385d2f683579e0bb20b3ea9404d179.

Vignesh-30 committed:
Updated to A coding standard

Vignesh-30 committed:
Changed string to Static string in Task class

Vignesh-30 committed:
Added a override print statement to task class

Vignesh-30 committed:
Added a Deadline class.

Vignesh-30 committed:
Added a Event class.

Vignesh-30 committed:
Added a Todo class.

Vignesh-30 committed:
Added Level 4.

Vignesh-30 committed:
renamed variable "by" to "dueBy"

Vignesh-30 committed:
Removed commented code.

Vignesh-30 committed:
Removed commented code.

Vignesh-30 committed:
Made constant variable final.
Extracted all commands and made them methods.
Improved code readability. A-CodeQuality

Vignesh-30 committed:
Fixed code formatting in event class

Vignesh-30 committed:
Commands mark, unmark, list checked for errors

Vignesh-30 committed:
Commands bye, todo, deadline, event checked for errors

Vignesh-30 committed:
If-else statements changed to switch statements

nghochi123 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,128 @@
+import java.util.Scanner;
+
 public class Duke {
+    static final String LINE = "\t____________________________________________________________";
```
Good that there are constants for different commands so that it is easy to change everything later on.

nghochi123 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,128 @@
+    static Task[] tasks = new Task[100];
+    static int textCount = 0;
+
+    public static void doCommandGreet() {
```
Constants and function names are standard and easy to understand.

nghochi123 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,128 @@
+    public static void handleUserCommand(String userCommand) {
+        String[] extractFirstWord = userCommand.split(" ", 2);
+        String firstWord = extractFirstWord[0];
+        if (firstWord.equals(COMMAND_MARK)) {
```
A switch statement would work as well.

nghochi123 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,128 @@
+            int indexOfEventDetailsPartOne = extractFirstWord[1].indexOf("/from");
+            int indexOfEventDetailsPartTwo = extractFirstWord[1].indexOf("/to");
+            String eventName = extractFirstWord[1].substring(0, indexOfEventDetailsPartOne);
+            String eventDetailsPartOne = extractFirstWord[1].substring(indexOfEventDetailsPartOne + 6, indexOfEventDetailsPartTwo - 1);
```
You might want to split this part into a second line as it's too long.

nghochi123 commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,14 @@
+    protected String eventDetailsPartOne;
+    protected String eventDetailsPartTwo;
+
+    public Event(String description, String eventDetailsPartOne, String eventDetailsPartTwo) {
```
eventDetailsPartOne and Two are quite ambiguous - variables named from and to (or variants) would be sufficient.

nghochi123 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,31 @@
+public class Task {
+    private boolean isDone;
+    private String taskName;
+    static String done = "[X] ";
```
The variables 'done' and 'notDone' are constants, and should be fully capitalized.

de-yi replied:
I like how you used "private" instead of public scope so that it can only be accessed by methods in this class:)

nghochi123 reviewed (commented):
Overall, done pretty well! Good job.

de-yi commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,128 @@
+        System.out.println("\t" + "Task added!");
+        System.out.println("\t  " + tasks[textCount - 1]);
+        System.out.println("\t" + "Now you have " + textCount + " pending tasks.");
+        System.out.println(LINE);
```
Since this part appears in several methods, it can be extracted and reused!

de-yi commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,128 @@
+    static final String COMMAND_UNMARK = "unmark";
+    static final String COMMAND_EVENT = "event";
+    static final String COMMAND_TODO = "todo";
+    static final String COMMAND_DEADLINE = "deadline";
```
I like how you made different user commands into constants and listed at the front.

de-yi commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,128 @@
+        for (int index = 0; index < textCount; index++) {
+            System.out.print("\t" + count + ".");
+            System.out.println(tasks[index]);
+            count++;
```
Just an idea but maybe you can use (index+1) instead of setting another variable named count!

de-yi commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,128 @@
+    static final String COMMAND_TODO = "todo";
+    static final String COMMAND_DEADLINE = "deadline";
+    static Task[] tasks = new Task[100];
+    static int textCount = 0;
```
Just a minor quention: Why is this variable named textCount instead of taskCount??

de-yi commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,31 @@
+        this.isDone = false;
+    }
+
+    public String Completed() {
```
Shouldn't this method use small c instead of capital C?

de-yi reviewed (commented):
Overall I liked how you tried to use OOP and fllow coding standards that we learnt during lecture :) Good luck with latter parts!

Vignesh-30 committed:
Merge branch 'branch-Level-5'

Vignesh-30 committed:
Added packages duke and duke.task

Vignesh-30 committed:
Merge branch 'branch-A-Packages'

Vignesh-30 committed:
refactored done and not_done variables in task class as a Final variable

Vignesh-30 committed:
Converted array of tasks to an ArrayList of tasks. Added support for deletion of tasks.

Vignesh-30 committed:
added exceptions to catch errors in ArrayList

Vignesh-30 committed:
Added method to print invalid number input to user.

Vignesh-30 committed:
Added methods to print invalid format message and empty command message to user

Vignesh-30 committed:
renames textCount to taskCount

Vignesh-30 committed:
added ability to save tasks whenever task list is changed.

Vignesh-30 committed:
format changes of message to be printed out

Vignesh-30 committed:
added returnCommand method to return userCommand format to save tasks

Vignesh-30 committed:
added code for duke to auto add tasks from saved file.

Vignesh-30 committed:
Merge branch 'branch-Level-6'

Vignesh-30 committed:
Merge branch 'branch-Level-7' and handled merge conflicts

# Conflicts:
#	src/main/java/duke/Duke.java

Vignesh-30 committed:
minor changes to endure code quality

Vignesh-30 committed:
added numbers for printing out stored tasks

Vignesh-30 committed:
fixed a bug in message printed to user for failed delete command

Vignesh-30 committed:
minor changes in message to user

Vignesh-30 committed:
Adjusted coding standard in task class

Vignesh-30 committed:
changed return value of returnCommand() in task classes

Vignesh-30 committed:
Duke now stores marked / unmarked status of tasks

Vignesh-30 committed:
Extracted out Ui class to deal with printing messages to user

Vignesh-30 committed:
Extracted out TaskList class to store all task commands like add, delete, mark/unmark etc

Vignesh-30 committed:
moved greet and bye commands from TaskList to Ui class

Vignesh-30 committed:
Added method to printInvalidCommand

Vignesh-30 committed:
made string in Ui class public to enable access

Vignesh-30 committed:
changes in taskList to accesss ArrayList of tasks

Vignesh-30 committed:
Extracted Storage class to create file, edit file, extract data from file

Vignesh-30 committed:
Extracted out Parser class to make sense of the user command

Vignesh-30 committed:
deleted changes

Vignesh-30 committed:
moved handle user command to tasklist package

Vignesh-30 committed:
line formatting

Vignesh-30 committed:
Created command class and moved all commands inside it

Vignesh-30 committed:
created parser, storage,ui and command packages

Vignesh-30 committed:
Changed doCommandGreet and doCommandBye into printGreetMessage and printByeMessage
Moved method to print loaded file into ui class

Vignesh-30 committed:
added command to find a task by typing a keyword

okkhoy commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,61 @@
+    public Parser(String userInput) {
+        this.userInput = userInput;
+    }
+    //public String[] extractFirstWord = userInput.split(" ", 2);
```
pl. don't retain commented code

okkhoy commented on src/main/java/duke/tasklist/TaskList.java:
```diff
@@ -0,0 +1,105 @@
+                Command.doCommandMark(taskNum);
+            } catch (IndexOutOfBoundsException | NumberFormatException out_mark_a) {
+                Ui.printInvalidNumber("mark");
+            }
```
Can extract these to separate methods and invoke the method here

okkhoy reviewed (commented):
In general, good code.
1. I don't see any tests in your codebase. please add them
2. Please add Javadoc comments to all nontrival methods/classes

Vignesh-30 committed:
Merge pull request #1 from Vignesh-30/branch-Level-9

merging branch 9 into master

Vignesh-30 committed:
Merge branch 'master' of https://github.com/Vignesh-30/ip

Vignesh-30 committed:
Revert "merging branch 9 into master"

Vignesh-30 committed:
Merge pull request #2 from Vignesh-30/revert-1-branch-Level-9

Revert "merging branch 9 into master"

Vignesh-30 committed:
Revert "Revert "merging branch 9 into master""

Vignesh-30 committed:
Merge pull request #3 from Vignesh-30/revert-2-revert-1-branch-Level-9

Revert "Revert "merging branch 9 into master""

Vignesh-30 committed:
removed commented lines

Vignesh-30 committed:
Merge remote-tracking branch 'origin/master'

Vignesh-30 committed:
added JavaDoc comments into classes and methods

Vignesh-30 committed:
Merge branch 'master' into branch-A-JavaDoc

# Conflicts:
#	src/main/java/duke/tasklist/TaskList.java

Vignesh-30 committed:
Merge pull request #4 from Vignesh-30/branch-A-JavaDoc

merging branch-A-JavaDoc into Master

Vignesh-30 committed:
Merge branch 'master' of https://github.com/Vignesh-30/ip

Vignesh-30 committed:
More JavaDoc comments added after megring of PRs

Vignesh-30 committed:
removed extra line

Vignesh-30 committed:
added curly braces for else block

Vignesh-30 committed:
seperate command into 5 classes,
Command stores commands to add todo, deadline and event tasks
4 more command classes for delete,mark,unmark and list commands

Vignesh-30 committed:
Formatting changes to add line

Vignesh-30 committed:
Final addition of comments

Vignesh-30 committed:
updated README

Vignesh-30 committed:
format changes to ensure code quality
